numpy/core/src/multiarray/datetime.c

```diff
@@ -3783,6 +3804,29 @@ recursive_find_object_timedelta64_type(PyObject *obj,
         else if (arr_dtype->type_num != NPY_OBJECT) {
             return 0;
         }
+        else {
+            Py_ssize_t ndim;
+            ndim = PyArray_NDIM((PyArrayObject *)obj);
```

Comment:
Blank line after declarations.
The return type of `PyArray_NDIM` is `int`, any reason to change that?

Reply:
I note that up above `arr` is already the cast of `obj`, so why not
```
            if (PyArray_NDIM(arr) == 0) {
```